src/react/controllers/map-controller.js

```diff
@@ -109,18 +113,122 @@ export default class MapController extends PureComponent {
 
   componentWillUpdate(nextProps) {
     this._controls.setOptions(nextProps);
+    this._animateViewportProp(nextProps);
   }
 
   componentWillUnmount() {
     this._eventManager.destroy();
   }
 
-  _onInteractiveStateChange({isDragging = false}) {
+  _onInteractiveStateChange(interactiveState) {
+    const {isDragging = false} = interactiveState;
     if (isDragging !== this.state.isDragging) {
       this.setState({isDragging});
     }
   }
 
+  _extractViewportFromProps(props) {
+    return {
+      width: props.width,
+      height: props.height,
+      latitude: props.latitude,
+      longitude: props.longitude,
+      zoom: props.zoom,
+      bearing: props.bearing,
+      pitch: props.pitch,
+      altitude: props.altitude
+    };
+  }
+
+  _animateViewportProp(nextProps) {
+    if (this.props.animateViewport !== true) {
+      return;
+    }
+    const startViewport = this._extractViewportFromProps(this.props);
+    const endViewport = this._extractViewportFromProps(nextProps);
+    if (this._viewportChanged(startViewport, endViewport)) {
+      if (this.state.animationInterval) {
+        clearInterval(this.state.animationInterval);
+      }
+      const animationInterval = setInterval(() => this._updateViewport(), 50);
+      this.setState({
+        animationInProgress: true,
+        animationT: 0.0,
+        animationStartViewport: startViewport,
+        animationEndViewport: endViewport,
+        animationInterval,
+        animatedViewport: startViewport
+      });
+    }
+  }
+
+  _viewportChanged(startViewport, endViewport) {
```

Comment:
A one-line comment here.  It is only looking for animated props, should the function name hint about that?
`didViewportAnimatePropsChange`?